modules/web/test/com/haulmont/charts/model/amcharts/charts/PivotTableSerializerTest.java

```diff
@@ -0,0 +1,168 @@
+
+    @ClassRule
+    public static TestUiEnvironment environment =
+            //Using this container to avoid fiddling with user session, authorization
+            new TestUiEnvironment(ChartsTestContainer.Common.INSTANCE) {
+                @Override
+                protected void cleanupEnvironment() {
+                    //we don't have windows, we don't have client cache
+                }
+            }
+            .withLocale(Locale.ENGLISH)
+            .withUserLogin("admin");
+
+    @Before
+    public void setupTest() {
+        dataItems = new ArrayList<>();
+        LocalDateTime localDateTime = LocalDateTime.of(2010, Month.APRIL, 10, 14, 11, 59);
+        Date date = Date.from(localDateTime.toInstant(ZoneOffset.UTC));
+        dataItems.add(new PivotDataItem(1L,
+                "one",
+                localDateTime,
+                localDateTime.toLocalDate(), date));
+    }
+
+    @Test
+    public void dateSerialziationTest() {
+
+        PivotTableDataItemsSerializer serializer = new PivotTableDataItemsSerializer();
+        serializer.setUserSessionSource(environment.getSessionSource());
+        PivotTableModel model = getPivotTableModel();
+        PivotJsonSerializationContext ctx = new PivotJsonSerializationContext(model, new Gson());
+        JsonArray serialize = serializer.serialize(dataItems, ctx);
+
+        JsonObject jsonObject = serialize.getAsJsonArray().get(0).getAsJsonObject();
+        JsonElement id = jsonObject.get("id");
+        JsonElement name = jsonObject.get("name");
+        JsonElement localDate = jsonObject.get("localDate");
+        JsonElement localDateTime = jsonObject.get("localDateTime");
+        JsonElement date = jsonObject.get("date");
+
+        //All values should be serialized
+        Assert.that(id.isJsonPrimitive(), "Long values should be serialized");
+        Assert.that(name.isJsonPrimitive(), "String values should be serialized");
+        Assert.that(localDate.isJsonPrimitive(), "LocalDate values should be serialized");
+        Assert.that(localDateTime.isJsonPrimitive(), "LocalDateTime values should be serialized");
+        Assert.that(date.isJsonPrimitive(), "Date values should be serialized");
+    }
+
+    private PivotTableModel getPivotTableModel() {
+        Map<String, String> props = new HashMap<>();
+        props.put("id", "id");
+        props.put("name", "name");
+        props.put("localDate", "localDate");
+        props.put("localDateTime", "localDateTime");
+        props.put("date", "date");
+
+        Aggregation aggregation = new Aggregation();
+        aggregation.setMode(AggregationMode.COUNT);
+
+        PivotTableModel model = new PivotTableModel();
+        model.setProperties(props);
+        model.addCols("localDate");
+        model.addRows("name");
+        model.addAggregationProperties("id");
+        model.setAggregation(aggregation);
+        return model;
+    }
+
+    static class PivotDataItem implements DataItem {
```

Comment:
Do we really need a separate class for a single-use? I suppose that `com.haulmont.charts.gui.data.MapDataItem` can fit your needs.

Reply:
Fair enough, will update